### ISSUE TITLE
bug: Step Functions fails to execute nested maps

### ISSUE THREAD

**Issue by guillaume-macquat** [localstack/localstack#10467]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I'm trying to execute a state machine using the following definition, which contains a nested map:
```json
{
  "StartAt": "SetupVariables",
  "States": {
    "SetupVariables": {
      "Type": "Pass",
      "Parameters": {
        "items": [
          [1, 2, 3],
          [4, 5, 6]
        ]
      },
      "Next": "Map1"
    },
    "Map1": {
      "Type": "Map",
      "ItemsPath": "$.items",
      "ItemProcessor": {
        "ProcessorConfig": { "Mode": "INLINE" },
        "StartAt": "Map2",
        "States": {
          "Map2": {
            "Type": "Map",
            "ItemProcessor": {
              "ProcessorConfig": { "Mode": "INLINE" },
              "StartAt": "Pass",
              "States": {
                "Pass": {
                  "Type": "Pass",
                  "End": true
                }
              }
            },
            "End": true
          }
        }
      },
      "End": true
    }
  }
}
```

The execution fails with the following error:
```
{
  "error": "States.Runtime",
  "cause": "An error occurred while executing the state 'Map1' (entered at the event id #41). cannot pickle '_thread.lock' object"
}
```

### Expected Behavior

The execution should succeed.

### How are you starting LocalStack?

With a docker-compose file

### Steps To Reproduce

Create and execute a state machine using the provided definition.

### Environment

```markdown
- LocalStack: 3.2.1.dev20240314122304
```

### Anything else?

I already reported this bug in the related issue #9809 but since it has been unanswered for a while I think it might be beneficial to create my own issue here.

**Comment by MEPalma** [comment]
@guillaume-macquat, thank you for assembling this bug report. Recent changes have been incorporated to address the issue you highlighted. These will be available in the next nightly version too. Your feedback on the effectiveness of these updates in resolving the reported issue would be greatly appreciated.

**Comment by guillaume-macquat** [comment]
I confirm this issue is solved in version 3.3.1.dev20240402202444.
Thank you very much @MEPalma and the LocalStack team
